src/lib/logger.test.ts

```diff
@@ -0,0 +1,172 @@
+    const document = DocumentModel.create({
+      type: SectionDocument,
+      uid: "1",
+      key: "source-document",
+      createdAt: 1,
+      content: {},
+      visibility: "public"
+    });
+    stores.documents.add(document);
+
+    mock.post(/.*/, (req, res) => {
+      const request = JSON.parse(req.body());
+
+      expect(request.event).toBe("CREATE_TILE");
+      // expect(request.parameters.objectId).toBe(tile.id);
+      expect(request.parameters.objectType).toBe("Text");
+      expect(request.parameters.serializedObject).toEqual({
+        type: "Text",
+        text: "test"
+      });
+      expect(request.parameters.documentKey).toBe("source-document");
+      expect(request.parameters.documentType).toBe("section");
+
+      done();
+      return res.status(201);
+    });
+
+    await document.content.addTextTile("test");
+  });
+
+  it("can log copying tiles between documents", async (done) => {
+    const sourceDocument = DocumentModel.create({
+      type: SectionDocument,
+      uid: "source-user",
+      key: "source-document",
+      sectionId: "source-section",
+      createdAt: 1,
+      content: {},
+      visibility: "public"
+    });
+    const content = createSingleTileContent({ type: "Text", text: "test" });
+    sourceDocument.setContent(DocumentContentModel.create(content));
+
+    const destinationDocument = DocumentModel.create({
+      type: SectionDocument,
+      uid: "destination-user",
+      key: "destination-document",
+      sectionId: "destination-section",
+      createdAt: 1,
+      content: {},
+      visibility: "public"
+    });
+
+    stores.documents.add(sourceDocument);
+    stores.documents.add(destinationDocument);
+
+    let tileToCopy: ToolTileModelType;
+
+    mock.post(/.*/, (req, res) => {
+      const request = JSON.parse(req.body());
+
+      expect(request.event).toBe("COPY_TILE");
+      // expect(request.parameters.objectId).toBe(tile.id);
+      expect(request.parameters.objectType).toBe("Text");
+      expect(request.parameters.serializedObject).toEqual({
+        type: "Text",
+        text: "test"
+      });
+      expect(request.parameters.documentKey).toBe("destination-document");
+      expect(request.parameters.documentType).toBe("section");
+      expect(request.parameters.objectId).not.toBe(tileToCopy.id);
+      expect(request.parameters.section).toBe("destination-section");
+      expect(request.parameters.sourceDocumentKey).toBe("source-document");
+      expect(request.parameters.sourceDocumentType).toBe("section");
+      expect(request.parameters.souceObjectId).toBe(tileToCopy.id);
+      expect(request.parameters.sourceUsername).toBe("source-user");
+      expect(request.parameters.sourceSection).toBe("source-section");
+
+      done();
+      return res.status(201);
+    });
+
+    // annoying way to get a tile...
+    const firstRow = sourceDocument.content.rowMap.get(sourceDocument.content.rowOrder[0]);
```

Comment:
Looks like a good argument for `DocumentContent` to have a `getRowByIndex()` view function.